packages/typespec-client-generator-core/src/package.ts

```diff
@@ -616,14 +616,11 @@ export function getSdkPackage<
   context.__clients = new Array<SdkClientType<TServiceOperation>>();
   populateApiVersionInformation(context);
   const modelsAndEnums = diagnostics.pipe(getAllModelsWithDiagnostics(context));
-  for (const client of listClients(context)) {
-    createSdkClientType(context, client);
-  }
   const crossLanguagePackageId = diagnostics.pipe(getCrossLanguagePackageId(context));
   return {
     name: getClientNamespaceString(context)!,
     rootNamespace: getClientNamespaceString(context)!,
-    clients: Array.from(context.__clients.values()),
+    clients: listClients(context).map((c) => diagnostics.pipe(createSdkClientType(context, c))),
```

Comment:
maybe we should remove context.__clients ? I don't see where this value being used somewhere else, maybe I miss something ?